### PR TITLE
feat: Add `toReadableStream()` method to `SseStream`

### LINKED ISSUE #1465: Expose raw ReadableStream from Orchestration streaming API
## Describe the Problem

When building AI SDK providers (e.g., Vercel AI SDK), the standard pattern for streaming is to use `ReadableStream.pipeThrough(TransformStream)` for efficient, memory-friendly stream transformation.

Currently, `OrchestrationClient.stream()` returns an `OrchestrationStreamResponse` with a `.stream` property that is an `AsyncIterable<OrchestrationStreamChunkResponse>`. This requires manual iteration with `for await...of` and creating a new `ReadableStream` via the `start()` callback pattern:

```typescript
const sdkStream = streamResponse.stream;

const transformedStream = new ReadableStream({
  async start(controller) {
    for await (const chunk of sdkStream) {
      controller.enqueue(transformChunk(chunk));
    }
    controller.close();
  }
});
```

This approach:
- Cannot leverage native `pipeThrough()` for composable transformations
- Requires buffering chunks in the `start()` callback instead of pull-based streaming
- Makes backpressure handling more complex

## Propose a Solution

Expose the raw `ReadableStream<Uint8Array>` from the HTTP response, allowing consumers to build their own parsing/transformation pipeline:

```typescript
interface OrchestrationStreamResponse<T> {
  // Existing
  stream: OrchestrationStream<T>;
  
  // New
  rawStream: ReadableStream<Uint8Array>;
}
```

Or alternatively, provide a method to get the stream before SSE parsing:

```typescript
const response = await client.stream(request);
const rawStream = response.getRawStream(); // ReadableStream<Uint8Array>

const transformed = rawStream
  .pipeThrough(new TextDecoderStream())
  .pipeThrough(new EventSourceParserStream())
  .pipeThrough(myCustomTransform);
```

## Describe Alternatives

1. **Use `convertAsyncIteratorToReadableStream`** - Works but uses `pull()` pattern which is less efficient than direct piping
2. **Keep current implementation** - Functional but not aligned with Web Streams API best practices

## Additional Context

Reference implementations from Vercel AI SDK providers (OpenAI, Anthropic, Google) all use the `pipeThrough(TransformStream)` pattern:

```typescript
// From @ai-sdk/openai
return {
  stream: response.pipeThrough(
    new TransformStream<ParseResult<Chunk>, LanguageModelV3StreamPart>({
      transform(chunk, controller) {
        controller.enqueue(transformedChunk);
      },
      flush(controller) {
        controller.enqueue({ type: 'finish', ... });
      }
    })
  )
};
```

This pattern enables composable, memory-efficient streaming with proper backpressure handling.

### PR DESCRIPTION
**Pull request by jerome-benoit**
## Summary

Adds a `toReadableStream()` method to `SseStream` in `@sap-ai-sdk/core`, enabling consumers to convert the async iterable stream into a Web Streams API `ReadableStream` for composable transformations via `pipeThrough()`.

Closes #1465

## Changes

- **`packages/core/src/stream/sse-stream.ts`** — Add `toReadableStream()` method to `SseStream<Item>` class
- **`packages/core/src/stream/sse-stream.test.ts`** — Add 5 tests covering NDJSON output, empty streams, cancel/abort propagation, and error handling
- **`.changeset/sse-stream-to-readable-stream.md`** — Minor changeset for `@sap-ai-sdk/core`

## Design

Follows the same pattern as [OpenAI](https://github.com/openai/openai-node/blob/03ee9c5f/src/core/streaming.ts#L167-L194) and [Anthropic](https://github.com/anthropics/anthropic-sdk-typescript/blob/fdaa72be/src/core/streaming.ts#L133-L155) SDKs:

- **Pull-based** `ReadableStream` with native backpressure handling — `pull()` is only called when the consumer is ready
- **NDJSON format** — each item is `JSON.stringify(value) + '\n'`, one item per line
- **Proper cleanup** — `cancel()` calls `iter.return?.()` and aborts the controller
- **Inherited by subclasses** — `OrchestrationStream` and `AzureOpenAiChatCompletionStream` get it for free

### Usage

```typescript
const response = await client.stream(request);

// Existing: AsyncIterable consumption
for await (const chunk of response.stream) { /* ... */ }

// New: Web Streams API consumption with pipeThrough()
const readable = response.stream.toReadableStream();
const transformed = readable.pipeThrough(myTransformStream);
```

### CODE REVIEW

**Review by davidkna-sap — COMMENTED** [review]
Thank you for opening the PR @jerome-benoit.

Regarding the implementation, have you considered using `ReadableStream.from`? In general, it also looks like stringification is not necessary to use `ReadableStream`.

File: .changeset/sse-stream-to-readable-stream.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@sap-ai-sdk/core': minor
+---
+
+[feat] Add `toReadableStream()` method to `SseStream`.
+The new method converts the async iterable stream to a pull-based `ReadableStream` of newline-delimited JSON, enabling composable transformations via the Web Streams API `pipeThrough()` pattern with native backpressure handling.
```
Comment (davidkna-sap) [inline .changeset/sse-stream-to-readable-stream.md]: [req] Fix vale errors:
```suggestion

[feat] Add `toReadableStream()` method to `SseStream`.
The new method converts the asynchronous iterable stream to a pull-based `ReadableStream` of newline-delimited JSON, enabling composable transformations via the Web Streams API `pipeThrough()` pattern with native flow control.
```

**Review by davidkna-sap — CHANGES_REQUESTED** [review]
Please resolve the spell-checker errors, I have made a suggestion.

We have discussed this and decided to accept the PR as-is once pipeline failures are addressed.

File: .changeset/sse-stream-to-readable-stream.md (hunk 1 above)
Comment (jerome-benoit) [reply .changeset/sse-stream-to-readable-stream.md]: Applied via suggestion, thanks. Branch is now synced with `main`.

**Review by davidkna-sap — APPROVED** [review]

**Comment by davidkna-sap** [comment]
Thanks for the PR @jerome-benoit!
